common/models/templates/src/trie-model.ts

```diff
@@ -81,6 +83,114 @@ type TextWithProbability = {
   p: number; // real-number weight, from 0 to 1
 }
 
+class Traversal implements LexiconTraversal {
```

Comment:
Why'd I move the inner class out of `TrieModel`?

For some odd reason, when it was within `TrieModel`, `TrieModel` as a whole would fail to treeshake out of `app/webview` - the code for the "main thread", as it were.  There's an inclusion of the `buildMergedTransform` that should only extract that one piece... but it also pulled `TrieModel` _and_ the wordbreaker implementations, complete with the backing Unicode data, due to the failed treeshaking.

This simple relocation was enough to allow `TrieModel` and the wordbreakers to treeshake out as appropriate, which made a 25-30 KB difference for the app/webview build artifact.  Unfortunately, I couldn't quite determine a minimal reproduction for this, so no bug report.

On the _other_ hand, I did find something else I _could_ minimal-repro:  `QuoteBehavior` does not treeshake out due to https://github.com/evanw/esbuild/issues/3555.   It's quite small though, so it's not a dealbreaker here.

I spend time on these issues because it _is_ odd how common/models/types needed special treeshaking handling in #10257; I figured there was a chance of correlation that could help clean things up there too.  No such luck yet, though.